torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -57,12 +55,61 @@ void propagateNode(Node* n) {
   auto graph = n->owningGraph();
   WithInsertPoint guard(n);
   for (size_t i = 0; i < outputs.size(); ++i) {
-    auto new_output = graph->insertConstant(outputs[i]);
+    auto new_output = insertConstant(*graph, outputs[i]);
     n->outputs()[i]->replaceAllUsesWith(new_output);
     // let dce elimination remove n
   }
 }
 
+void inlineIf(Block *body, Node * n) {
+  for(auto it = body->nodes().begin(); it != body->nodes().end();) {
+    Node *body_node = *it;
+    //advance iterator because after body_node is moved its next pointer will be
+    //to n
+    it++;
+    body_node->moveBefore(n);
+  }
+  for (size_t i = 0; i < n->outputs().size(); ++i) {
+    n->outputs().at(i)->replaceAllUsesWith(body->outputs().at(i));
+  }
+  // NB: destroy the node here, because it might contain side effects, like print
+  n->destroy();
+}
+
+bool isTrueConstant(Value *val) {
+  at::optional<bool> maybe_value = constant_as<bool>(val);
+  return maybe_value && *maybe_value;
```

Comment:
Perhaps we should assert that it's constant here? Otherwise `inlineIf` could mistakenly take the `else` branch if the value is not a constant.